### PR TITLE
HPCC-29629 Provide Grafana Loki-Stack startall opt

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds option to deploy managed loki stack
- Adds stopall directive to stop loki
- Sets HPCC format to JSON if Loki deployed
- Fixes minor/unrelated tab issue

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dockerfiles/startall.sh [hunk 1]
```diff
@@ -129,6 +133,9 @@ while [ "$#" -gt 0 ]; do
       m) DEPLOY_PROM=1
          PROMETHEUS_METRICS_SINK_ARG="-f $scriptdir/../helm/examples/metrics/prometheus_metrics.yaml"
          ;;
+      g) DEPLOY_LOKI=1
+         LOG_FORMAT_JSON_ARG="--set global.logging.format="json""
```
Comment (rpastrana) [inline dockerfiles/startall.sh:137]: Not expecting to keep this, but it illustrates how we can declare most advantageous log format for the expected back-end log processor....

File: dockerfiles/startall.sh [hunk 2]
```diff
@@ -191,6 +198,21 @@ if [[ $DEPLOY_PROM == 1 ]] ; then
   helm ${CMD} myprometheus4hpccmetrics $scriptdir/../helm/$MANAGED_PROM_SUBPATH ${restArgs[@]} --set kube-prometheus-stack.prometheus.service.type=LoadBalancer --set kube-prometheus-stack.grafana.service.type=LoadBalancer
 fi
 
+if [[ $DEPLOY_LOKI == 1 ]] ; then
+  echo -e "\n\nDeploying "myloki4hpcclogs" - Grafana Loki Stack:"
+  echo -e "\n Access Grafana on: http://localhost:3000"
+  echo -e "\n user: admin"
+  echo -e "\n To obtain pass: 'kubectl get secret myloki4hpcclogs-grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo' "
+  echo -e "\n Find logs on 'Explore' view, under 'loki' datasource\n"
```
Comment (rpastrana) [inline dockerfiles/startall.sh:206]: New concept in this script, but ultimately very useful, and very low negative impact

File: dockerfiles/stopall.sh [hunk 3]
```diff
@@ -34,11 +34,11 @@ while [ "$#" -gt 0 ]; do
          CLUSTERNAME=$1
          ;;
       -e) UNINSTALL_ELK=0
-	     echo "elastic4hpcclogs will not be stopped..."
+       echo "elastic4hpcclogs will not be stopped..."
```
Comment (rpastrana) [inline dockerfiles/stopall.sh:37]: Minor and unrelated to target Jira, but ultimately will not confuse anybody and resolves minor annoyance without spending unnecessary developer time fixing it in dedicated jira.

Comment (ghalliday) [inline dockerfiles/stopall.sh:37]: The corrected indent doesn't look correct.

File: dockerfiles/startall.sh (hunk 1 above)
Comment (ghalliday) [reply dockerfiles/startall.sh:137]: What happens if this is not specified?

**Review by ghalliday — COMMENTED** [review]
Sensible functionality

File: dockerfiles/startall.sh (hunk 1 above)
Comment (rpastrana) [reply dockerfiles/startall.sh:137]: no-op, like the other uninitialized vars, however, I'm renaming to LOG_FORMAT_ARG and defaulting to CSV.

**Review by ghalliday — APPROVED** [review]
@rpastrana please squash and I will merge

**Comment by rpastrana** [comment]
> @rpastrana please squash and I will merge

Done.

**Pull request merged**